### ISSUE TITLE
large_enum_variant should not lint types that are copy

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#8894]
### Summary

````rust
#[derive(Copy, Clone)]
enum Test {
    A(i32),
    B([i32; 8000]),
}
````
the lint will fire here, but it seems that `Box` does not impl the `Copy` trait...?

````
error[E0204]: the trait `Copy` may not be implemented for this type
 --> src/main.rs:1:10
  |
1 | #[derive(Copy, Clone)]
  |          ^^^^
...
4 |     B(Box<[i32; 8000]>),
  |       ---------------- this field does not implement `Copy`
  |
````

### Lint Name

large_enum_variant

**Comment by llogiq** [comment]
@rustbot  claim

I don't thiink the lint should completely avoid enums which are `Copy`, but should avoid giving the incorrect suggestion, noting that the `Copy` impl precludes introducing indirections.

**Comment by llogiq** [comment]
Options include removing the `Copy` bound or moving the data to an auxiliary structure (e.g. ECS, arena, etc.)
